tools/lsp/preview/native.rs

```diff
@@ -387,7 +387,13 @@ pub fn update_highlight(url: Option<Url>, offset: u32) {
         });
 
         if let Some(handle) = handle {
-            let path = url.as_ref().map(|u| u.to_string().into()).unwrap_or_default();
+            let Some(url) = &url else {
+                return;
+            };
+            let Ok(path) = url.to_file_path() else {
+                return;
+            };
```

Comment:
I'd have used as_ref().map() to save 3 lines.  But that's just nitpicking
```suggestion
            let Some(path) = url.as_ref().map(|u| u.to_file_path) else {
                return;
            };
```